contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -423,6 +423,11 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
             _enterDefault(pool, borrower, DefaultReason.InterestOverdue);
             return;
         }
+        uint256 overBorrowLimit = overBorrowLimitTime[pool][borrower];
+        if (overBorrowLimit != 0 && defaultTime >= overBorrowLimit) {
+            _enterDefault(pool, borrower, DefaultReason.LimitExceeded);
```

Comment:
I think we also need to check if it's above the limit again

Reply:
or maybe just reset the timer whenever borrower repays below the limit

Reply:
actually it's done it this PR so it doesn't seem to be a problem